frontend/types/hashTabs.ts

```diff
@@ -1,5 +1,12 @@
+import type { IconDefinition } from '@fortawesome/fontawesome-svg-core'
+import type { Component } from 'vue'
+
 export type HashTab = {
+  component?: Component,
   disabled?: boolean,
-  index: number,
+  icon?: IconDefinition,
+  index: string,
```

Comment:
```suggestion
  index: number,
```
this should stay a number

Reply:
renamed to key:string and refactored the whole thing once more